libpromises/eval_context.c

```diff
@@ -4029,35 +4021,51 @@ void EvalContextProfilingEnd(EvalContext *ctx, const Policy *policy)
     }
 
     int64_t end = EvalContextEventStart();
-    int64_t start = ctx->profiler.elapsed;
+    int64_t start = ctx->elapsed;
     assert(end >= start); /* sanity check */
 
-    ctx->profiler.elapsed = end - start;
-
-    HashMap *map = SumEventFrames(ctx->profiler.events);
-    JsonElement *profiling = JsonObjectCreate(2);
-
-    JsonElement *json_policy = PolicyToJson(policy);
-    JsonObjectAppendObject(profiling, "policy", json_policy);
+    ctx->elapsed = end - start;
 
     JsonElement *events = JsonArrayCreate(10);
     {
-        HashMapIterator iter = HashMapIteratorInit(map);
-        MapKeyValue *mkv;
-        while ((mkv = HashMapIteratorNext(&iter)) != NULL)
+        for (size_t i = 0; i < SeqLength(ctx->events); i++)
         {
-            EventFrame *event = mkv->value;
+            EventFrame *event = SeqAt(ctx->events, i);
             JsonArrayAppendObject(events, EventToJson(event));
         }
     }
-    JsonObjectAppendArray(profiling, "events", events);
-
     // write
     Writer *writer = FileWriter(stdout);
-    JsonWrite(writer, profiling, 2);
+    JsonWrite(writer, events, 2);
     WriterClose(writer);
 
-    JsonDestroy(profiling);
+    JsonDestroy(events);
+}
+
+EventFrame *EvalContextGetLastEventFrame(EvalContext *ctx)
+{
+    StackFrame *frame = LastStackFrameByEvent(ctx);
+
+    if (frame == NULL)
+    {
+        return NULL;
+    }
+
+    return frame->event;
```

Comment:
Consider using ternary operator
```suggestion
    return (frame == NULL) ? NULL : frame->event;
```